docs/hugo/content/architecture-decision-records/ADR-2022-02-Backward-Compatibility.md

```diff
@@ -1,17 +1,88 @@
+  Mitigation would require deleting the existing resource and creating a new CR with an updated version for the resource.
+* Existing automated deployments (e.g. as used by teams with a GitOpts model or other forms of continuous deployment) would break because the version specified in the existing YAML files would no longer exist.  
+  Mitigation would require modification of the deployment pipelines to change the version of the resources being deployed.
+* Existing manual deployments would also break because the version specified in the existing YAML files would no longer exist.
+  Mitigation would require modification of the YAML files being deployed to change the version of the resources being deployed.
+
+## Option 3a: In-place upgrade for existing resources
+
+Change the generated prefix to `v1beta` for all resources, but retain the alpha storage variants of each existing resource. This would allow an upgraded ASO installation to promote any existing resources to the new storage version.
+
+Pros
+
+* Moderate effort to implement (requires a new pipeline stage to create the alpha storage variants for compatibility with older versions).
+* Existing clusters with custom resources reliant on ASO would continue to run because ASO would support conversion from the old hub version to the new.
+
+Cons
+
+* Existing automated deployments (e.g. as used by teams with a GitOpts model or other forms of continuous deployment) would break because the version specified in the existing YAML files would no longer exist.  
+  Mitigation would require modification of the deployment pipelines to change the version of the resources being deployed.
+* Existing manual deployments would also break because the version specified in the existing YAML files would no longer exist.
+  Mitigation would require modification of the YAML files being deployed to change the version of the resources being deployed.
+
+## Option 3b: Retain the alpha version for existing resources
+
+Use a beta prefix for all resources. Also use our existing alpha prefix for all resources that were included in any alpha release of ASO, but not for new resources.
+
+Pros
+
+* Moderate effort to implement (requires pipeline and configuration changes to generate multiple versions of some resources).  
+  Could be implemented in two parts: generate all resources with both prefixes for `codegen-beta-0`, and introduce configuration for newer resources in `codegen-beta-1`.
+* Existing clusters with custom resources reliant on ASO would continue to run because ASO would support conversion from the old hub version to the new.
+* Existing automated deployments (e.g. as used by teams with a GitOpts model or other forms of continuous deployment) would continue to run because ASO would still support the version.
+* Existing manual deployments would also continue to run because ASO would still support the version.
+
+Cons
+
+* The size of our CRDs would double - and they're already large.  
```

Comment:
Is this actually a problem? I know there's a limit but it's only 25% more content than Option 3b, not double 3b?

Reply:
I've clarified the context. I don't think we're too close to the limit at the moment, but we can't afford to double very many times!